samples/azure/native-integration-asq/ASQN_9/Receiver/Program.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.Azure.Transports.WindowsAzureStorageQueues;
+using NServiceBus.Features;
+
+class Program
+{
+    static async Task Main()
+    {
+        var endpointName = "native-integration-asq";
+        Console.Title = endpointName;
+
+        var endpointConfiguration = new EndpointConfiguration(endpointName);
+        endpointConfiguration.UseSerialization<NewtonsoftSerializer>();
+        endpointConfiguration.EnableInstallers();
+
+        var transport = endpointConfiguration.UseTransport<AzureStorageQueueTransport>();
+        transport.ConnectionString("UseDevelopmentStorage=true");
+        transport.DisablePublishing();
+        transport.DelayedDelivery().DisableDelayedDelivery();
+
+        #region Native-message-mapping
+
+        transport.UnwrapMessagesWith(message => new MessageWrapper
+        {
+            Id = message.MessageId,
+            Body = message.Body.ToArray(),
+            Headers = new Dictionary<string, string>
+            {
+                { "NServiceBus.EnclosedMessageTypes", typeof(NativeMessage).FullName }
```

Comment:
Should we directly use the headers constant given that we reference NSB already?